content/en/blog/_posts/2021-07-16-read-write-once-pod-access-mode-alpha.md

```diff
@@ -0,0 +1,267 @@
+
+## What are access modes and why are they important?
+
+When using storage, there are different ways we can model how it is consumed.
+
+For example, a storage system like a network file share can have many users all reading and writing data simultaneously.
+In other cases maybe everyone is allowed to read data but not write it.
+For highly sensitive data, maybe only one user is allowed to read and write data but nobody else.
+
+In the world of Kubernetes, [access modes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) are the way we model how storage is consumed.
+These access modes are a part of the spec for PersistentVolumes (PVs) and PersistentVolumeClaims (PVCs).
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: shared-cache
+spec:
+  accessModes:
+  - ReadWriteMany # Allow many pods to access shared-cache simultaneously.
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: ""
+```
+
+In Kubernetes, we currently have three access modes for PVs and PVCs:
+
+- ReadWriteOnce -- the volume can be mounted as read-write by a single node
+- ReadOnlyMany -- the volume can be mounted read-only by many nodes
+- ReadWriteMany -- the volume can be mounted as read-write by many nodes
+
+These access modes are enforced by Kubernetes components like the `kube-controller-manager` and `kubelet` to ensure only certain pods are allowed to access a given PersistentVolume.
+
+## What is this new access mode and how does it work?
+
+In Kubernetes v1.22, a fourth access mode for PVs and PVCs is introduced for CSI volumes:
+
+- ReadWriteOncePod -- the volume can be mounted as read-write by a single pod
+
+When a user creates a pod with a PVC that has the ReadWriteOncePod access mode, Kubernetes will ensure that pod is the only pod in the cluster that can read and write to it.
+
+If you create another pod that references the same PVC with this access mode, the pod will fail to start like so:
+
+```
+Events:
+  Type     Reason            Age   From               Message
+  ----     ------            ----  ----               -------
+  Warning  FailedScheduling  1s    default-scheduler  0/1 nodes are available: 1 node has pod using PersistentVolumeClaim with the same name and ReadWriteOncePod access mode.
+```
+
+### How is this different than the ReadWriteOnce access mode?
+
+The ReadWriteOnce access mode restricts volume access to a single *node*, which means it is possible for multiple pods on the same node to read from and write to the same volume.
+This could potentially be a major problem for some applications, especially if they require at most one writer for data safety guarantees.
+
+With ReadWriteOncePod these issues go away.
+Just set the access mode on your PVC, and Kubernetes will guarantee only a single pod has access.
```

Comment:
For technical writing, I try to avoid “just” in this sense - and especially when there _are_ extra steps, such as enabling feature gates across multiple components.

```suggestion
With ReadWriteOncePod these issues go away:
set the access mode on your PVC, and Kubernetes guarantees that only a single pod has access.
```